src/middlewared/middlewared/plugins/vm/usb.py

```diff
@@ -107,3 +107,13 @@ async def usb_passthrough_choices(self):
             mapping[device] = details
 
         return mapping
+
+    @private
+    async def get_usb_port_from_usb_details(self, usb_data):
+        if any(not usb_data.get(k) for k in ('product_id', 'vendor_id')):
```

Comment:
It's a sanity check and a private method - so if at some stage later some changes are introduced, it would fail hard and fast. I can remove it if you think we should.